src/Poseidon/CLI/Server.hs

```diff
@@ -0,0 +1,249 @@
+            return $ ServerApiReturnType [] (Just retData)
+
+        get "/groups" . conditionOnClientVersion $ do
+            let retData = ApiReturnGroupInfo . getAllGroupInfo $ allPackages
+            return $ ServerApiReturnType [] (Just retData)
+
+        get "/individuals" . conditionOnClientVersion $ do
+            maybeAdditionalColumnsString <- (Just <$> param "additionalJannoColumns") `rescue` (\_ -> return Nothing)
+
+            let (indInfo, pacVersions, additionalColumnEntries) = case maybeAdditionalColumnsString of
+                    Just additionalColumnsString ->
+                        let additionalColumnNames = splitOn "," additionalColumnsString
+                        in  getExtendedIndividualInfo allPackages additionalColumnNames
+                    Nothing -> getExtendedIndividualInfo allPackages []
+            let retData = ApiReturnIndividualInfo indInfo pacVersions additionalColumnEntries
+            return $ ServerApiReturnType [] (Just retData)
+
+        -- API for retreiving package zip files
+        unless (null zipDict) . get "/zip_file/:package_name" $ do
+            packageName <- param "package_name"
+            maybeVersionString <- (Just <$> param "package_version") `rescue` (\_ -> return Nothing)
+            maybeVersion <- case maybeVersionString of
+                Nothing -> return Nothing
+                Just versionStr -> case parseVersionString versionStr of
+                    Nothing -> raise . pack $ "Could not parse package version string " ++ versionStr
+                    Just v -> return $ Just v
+            case sortOn (Down . snd . fst) . filter ((==packageName) . fst . fst) $ zipDict of
+                [] -> raise . pack $ "unknown package " ++ packageName
+                [((_, pv), fn)] -> case maybeVersion of
+                    Nothing -> file fn
+                    Just v -> if pv == Just v then file fn else raise . pack $ "Package " ++ packageName ++ " is not available for version " ++ showVersion v
+                pl@((_, fnLatest) : _) -> case maybeVersion of
+                    Nothing -> file fnLatest
+                    Just v -> case filter ((==Just v) . snd . fst) pl of
+                        [] -> raise . pack $ "Package " ++ packageName ++ "is not available for version " ++ showVersion v
+                        [(_, fn)] -> file fn
+                        _ -> error "Should never happen" -- packageCollection should have been filtered to have only one version per package
+        notFound $ raise "Unknown request"
+
+-- this serves as a point to broadcast messages to clients. Adapt in the future as necessary.
+genericServerMessages :: [String]
+genericServerMessages = ["Greetings from the Poseidon Server, version " ++ showVersion version]
+
+-- this needs to be adapted in the future to mark the supported client versions
+minimalRequiredClientVersion :: Version
+minimalRequiredClientVersion = makeVersion [1, 1, 8, 5]
```

Comment:
Maybe this should live in `Poseidon.PoseidonVersion`

Reply:
Done.